lib/Doctrine/DBAL/Platforms/Keywords/MySQLKeywords.php

```diff
@@ -15,6 +15,15 @@ public function getName()
         return 'MySQL';
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function isKeyword($word)
+    {
+        return parent::isKeyword($word) ||  preg_match('/^\d+$/', $word) ||
+            preg_match('/[^\w$\x{0080}-\x{FFFF}]+/u', $word);
```

Comment:
The fact that an object name contains certain characters doesn't make it a keyword.

On MySQL, we can quote all identifiers. Whether or not it could be done without an API change, is a bigger question.

Please see https://github.com/doctrine/dbal/issues/4772 for more details.